llvm/lib/ObjectYAML/ELFEmitter.cpp

```diff
@@ -1799,6 +1799,21 @@ void ELFState<ELFT>::writeSectionContent(Elf_Shdr &SHeader,
   if (!Section.Notes)
     return;
 
+  unsigned Align;
+  switch (SHeader.sh_addralign) {
```

Comment:
Does using this work if `ShAddrAlign` is specified in the YAML to a value other than 0/4/8? I wasn't sure the order this was applied to the Elf_Shdr struct, in relation to this code. That field shouldn't have any functional impact on how the ELF is written, except to modify the value written in the section header (only `AddressAlign` should impact the way the section is written).

Reply:
Thanks, I've changed it to `Section.AddressAlign`.